frb_codegen/src/library/integration/utils.rs

```diff
@@ -4,30 +4,47 @@ use log::debug;
 use std::fs;
 use std::path::{Path, PathBuf};
 
-// ref: `Dir::extract`
+use crate::utils::path_utils::path_to_string;
+
+/// Recursively copies the file system structure starting at `current_reference_dir`
+/// to `base_target_path` while filtering by `filter` and invoking `modifier` on each file.
 pub(super) fn extract_dir_and_modify(
```

Comment:
hmm, it would be great to make this specific function unaware of things like rust crate dir (i.e. unaware of things in ReplaceContentConfig). Otherwise this one function is doing two things.

I guess you are doing this to avoid repeating the logic of replace rust crate dir etc - that's great! Maybe we can do it like:

* keep extract_dir_and_modify like the original one (i.e. only do one thing)
* create sth like `fn extract_dir_and_modify_that_is_aware_of_our_need_to_replace_some_variables()`, this one calls the extract_dir_and_modify, and this one is aware of the rust crate name logic etc.

Reply:
I see. For that specific question, maybe pass `compute_effective_path` as an *argument* may help.